designs/cloudstack-multiple-endpoints.md

```diff
@@ -131,20 +129,36 @@ for availabilityZone in availabilityZones:
 
 ### Cloudstack credentials
 
-
-In a multi-endpoint Cloudstack cluster, each endpoint may have its own credentials. We propose that Cloudstack credentials will be passed in via environment variable in the same way as they are currently,
-only as a list corresponding to AvailabilityZones. Currently, these credentials are passed in via environment variable, which contains a base64 encoded .ini file that looks like
-
-```
-[Global]
-api-key    = redacted
-secret-key = redacted
-api-url    = http://172.16.0.1:8080/client/api
+In a multi-endpoint Cloudstack cluster, each endpoint may have its own credentials.
+
+We propose that we make the leap to yaml k8s secrets for endpoint credentials of the form:
```

Comment:
I'm not clear on what you're asking for here. Just that the input file format to EKS-A change? Or should EKS-A also apply these Secrets to the cluster at some point? 

If you're asking for the latter:

1. how would CAPC know which secrets to use for which CloudstackFailureDomain?
2. would CAPC support a gradual migration to using these secrets by supporting *both* an environment variable input, as well as looking for secrets on the cluster?

Reply:
> Just that the input file format to EKS-A change? 
This and

> should EKS-A also apply these Secrets to the cluster at some point?
this.

Though I'm not sure EKS-A would need to explicitly apply the secrets file or if it could be inserted via kustomize.

Currently the secret is applied via clusterctl via EKS-A from an expanded environment variable against the CAPC manifest. (whew!)

While I would like it if the end user could run kustomize on the CAPC config directory to generate their own CAPC manifest, that probably gives the users a bit too much of a foot gun for EKS-A's liking. Barring that, EKS-A would probably need to apply the secrets.